### PR TITLE
Tests for `nndeploy::dag::Graph` and `nndeploy::dag::Edge`

### LINKED ISSUE #116: In DAG module, Graph should not be able to add itself as Node
### Describe the bug
added the test below and it passes. Ideally, `Graph` should not be able to add *itself* as a `Node`. This can lead to undefined behavior when processing `Node`s of such a `Graph`. 
```C++
TEST_F(GraphTest, GraphWithNodeAsItself) {
    auto inputs = std::vector<Edge *>();
    auto outputs = std::vector<Edge *>();
    auto edge_in =  std::make_unique<Edge>("edge_in");
    auto edge_out =  std::make_unique<Edge>("edge_out");
    inputs.emplace_back(edge_in.get());
    outputs.emplace_back(edge_out.get());
    auto graph = ConstructGraphWithVecArgs("@@!!##$$", inputs, outputs);
    ASSERT_TRUE(graph->addNode(graph.get()) == kStatusCodeOk); //this can lead to problems
}
```

### Potential fix
A mechanism within the `nndeploy::dag::Graph` to prevent this. Perhaps, something among lines of `if (node == this) return nullptr` in `nndeploy::base::Status nndeploy::dag::Graph::addNode(Node *node, bool is_external)` and `*nndeploy::dag::Graph::createInfer(...)`

### PR DESCRIPTION
**Pull request by Arpan3323**
### Describe the PR
This PR introduces the following changes:
- GoogleTest is added as a git submodule
- Tests for some functions in `nndeploy::dag::Graph` and `nndeploy::dag::Edge`
- Attempts to fix #116 and #117 in commit [5d67880](https://github.com/nndeploy/nndeploy/commit/5d6788047bc61a10cec23a03947e74d95d061747)

### Remarks
- There are still some functions, such as `nndeploy::dag::Graph::construct()`, `nndeploy::dag::Graph::init()`, etc. that are not tested. I will be adding tests for them shortly.
- The test linked below is failing: https://github.com/nndeploy/nndeploy/blob/aa13ce1df354b1d7aabc733f3d3691416c4a6500/test/source/nndeploy/dag/DAG_graph_test.cc#L117

### CODE REVIEW

File: test/source/nndeploy/dag/DAG_edge_test.cc
Comment (Alwaysssssss) [inline test/source/nndeploy/dag/DAG_edge_test.cc:1]: Thank you very much for adding the test module. Currently, the overall style of nndeploy follows the Google style. There are one or two adjustments based on this (for example, function names use lower camel case). For the current file named DAG_xxx_test, could it be renamed to dag_xxx_test?

**Review by Alwaysssssss — COMMENTED** [review]
Thank you very much for adding the test module. Currently, the overall style of nndeploy follows the Google style. There are one or two adjustments based on this (for example, function names use lower camel case). For the current file named DAG_xxx_test, could it be renamed to dag_xxx_test?

**Comment by Arpan3323** [comment]
Yes. I will rename the test files and keep this in mind for future. The cmake.config within the test directory has a function that searches for relevant test files based on module names, I will update that as well.

File: test/source/nndeploy/dag/DAG_edge_test.cc
Comment (Arpan3323) [reply test/source/nndeploy/dag/DAG_edge_test.cc:1]: renamed.

**Pull request merged**